### ISSUE TITLE
Saved Model gives  garbage prediction

### ISSUE THREAD

**Issue by ishmnnit** [horovod/horovod#3966]
**Environment:**
1. Framework: (TensorFlow, Keras, PyTorch, MXNet) Tensorflow
2. Framework version: 2.13
3. Horovod version: 0.28.1
8. Spark / PySpark version:  3.4
11. GCC version: 9

I am using the following to  Save a DCN  model - https://www.tensorflow.org/recommenders/examples/dcn in a Databricks cluster.

```
if hvd.rank() == 0:
      callbacks.append(ModelCheckpoint(filepath=checkpoint_filepath +'/{epoch:02d}',save_freq='epoch'))
return callbacks
```

Following to load the model

```
model = tf.keras.models.load_model(model_path, custom_objects={
"BinaryCrossentropy": tf.keras.losses.BinaryCrossentropy(from_logits=False, reduction=tf.keras.losses.Reduction.AUTO,
label_smoothing=0.0, axis=-1)
})
```

At training time all the metrics AUC, log loss look good but when I am doing prediction on the same validation and training dataset using the saved model, it’s giving me all garbage predictions with 0.5 AUC. The same code works fine without distributed training. 

Tensorflow -2.13
Tensorflow recommenders - 0.7.3
Horovod version - 0.28.1

**Comment by ishmnnit** [comment]
not a buf
